contributors/design-proposals/admission_control_extension.md

```diff
@@ -0,0 +1,594 @@
+
+##### Kubernetes
+
+Name | Code | Description
+---- | ---- | -----------
+ServiceAccount | serviceaccount/admission.go | Bind mount the service account token for a pod into the pod at a specific location.
+PersistentVolumeLabel | persistentvolume/label/admission.go | Lazily bind persistent volume claims to a given zone when a pod is scheduled.
+DefaultStorageClass | storageclass/default/admission.go | Set a default storage class on any PVC created without a storage class.
+
+Many other controllers have been proposed, including but not limited to:
+
+* ServiceInjectionPolicy to inject environment, configmaps, and secrets into pods that reference those services
+* Namespace level environment injection (all pods in this namespace should have env var `ENV=PROD`)
+* Label selector based resource defaults (all pods with these labels get these default resources)
+
+
+#### Referential Consistency
+
+These controllers enforce that certain guarantees of the system related to integrity.
+
+##### Kubernetes
+
+Name | Code | Description
+---- | ---- | -----------
+NamespaceAutoProvision | namespace/autoprovision/admission.go | When users create resources in a namespace that does not exist, ensure the namespace is created so it can be seen with `kubectl get namespaces`
+NamespaceExists | namespace/exists/admission.go | Require that a namespace object exist prior to a resource being created.
+NamespaceLifecycle | namespace/lifecycle/admission.go | More powerful and flexible version of NamespaceExists.
+
+##### OpenShift
+
+Name | Code | Description
+---- | ---- | -----------
+JenkinsBootstrapper | jenkinsbootstrapper/admission.go | Spawn a Jenkins instance in any project where a Build is defined that references a Jenkins pipeline. Checks that the creating user has permission to act-as an editor in the project to prevent escalation within a namespace.
+ImagePolicy | imagepolicy/imagepolicy.go | Performs policy functions like ImagePolicyWebhook, but also is able to mutate the image reference from a tag to a digest (fully qualified spec), look up additional information about the image from the OpenShift Image API and potentially enforce resource consumption or placement decisions based on the image. May also be used to deny images from being used that don't resolve to image metadata that OpenShift tracks.
+OriginNamespaceLifecycle | lifecycle/admission.go | Controls accepting resources for namespaces.
+
+
+### Patterns
+
+In a study of all known admission controllers, the following patterns were seen most often:
+
+1. Defaulting on creation
+2. Synchronous validation on creation
+3. Synchronous validation on update - side-effect free
+
+Other patterns seen less frequently include:
+
+1. Defaulting on update
+2. Resolving / further specifying values on update (ImagePolicy)
+3. Creating resources in response to user action with the correct permission check (JenkinsBootstrapper)
+4. Policy decisions based on *who* is doing the action (JenkinsBootstrapper)
```

Comment:
Also OwnerReferencesPermissionEnforcement, PodSecurityPolicy ( any admission plugin that implements `WantsAuthorizer`)